bin/check_partition_sizes.py

```
@@ -0,0 +1,418 @@
+
+def fetch_board_manifests(url, dest_dir):
+    """Download a platform archive and extract only its boards/*.json, as data.
+
+    Nothing from the archive is imported or executed, and every file is written to
+    dest_dir under its basename alone, so a crafted member name cannot escape it.
+    """
+    problem = check_manifest_url(url)
+    if problem:
+        raise ValueError(f"refusing to fetch board manifests: {problem}")
+
+    request = urllib.request.Request(url, headers={"User-Agent": "meshtastic-firmware-ci"})
```

Ruff (0.16.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)